src/citrine/informatics/predictors/mean_property_predictor.py

```diff
@@ -1,18 +1,26 @@
-from typing import List, Optional, Mapping
+from typing import List, Optional, Mapping, Union
 
 from citrine._rest.resource import Resource, ResourceTypeEnum
 from citrine._serialization import properties as _properties
 from citrine.informatics.data_sources import DataSource
-from citrine.informatics.descriptors import FormulationDescriptor, RealDescriptor
+from citrine.informatics.descriptors import (
+    FormulationDescriptor, RealDescriptor, CategoricalDescriptor, Descriptor
+)
 from citrine.informatics.predictors import Predictor
 from citrine._rest.ai_resource_metadata import AIResourceMetadata
 
 __all__ = ['MeanPropertyPredictor']
 
 
-class MeanPropertyPredictor(
-        Resource['MeanPropertyPredictor'], Predictor, AIResourceMetadata):
-    """A predictor interface that computes mean component properties.
+class MeanPropertyPredictor(Resource['MeanPropertyPredictor'], Predictor, AIResourceMetadata):
+    """A predictor that computes a component-weighted mean of real or categorical properties.
+
+    Each component in a formulation contributes to the mean property
+    a weight equal to its quantity raised to the `p` power.
```

Comment:
```suggestion
    a weight equal to its quantity raised to the power `p`.
```
I think this is the more common turn of phrase.